readthedocs/embed/v3/views.py

```diff
@@ -0,0 +1,318 @@
+            selector = f'#{fragment}'
+            node = HTMLParser(page_content).css_first(selector)
+        else:
+            html = HTMLParser(page_content)
+            node = self._find_main_node(html)
+
+        if not node:
+            return
+
+        if doctool == 'sphinx':
+            # Handle ``dt`` special cases
+            if node.tag == 'dt':
+                if 'glossary' in node.parent.attributes.get('class'):
+                    # Sphinx HTML structure for term glossary puts the ``id`` in the
+                    # ``dt`` element with the title of the term. In this case, we
+                    # return the parent node which contains the definition list
+                    # and remove all ``dt/dd`` that are not the requested one
+
+                    # Structure:
+                    # <dl class="glossary docutils">
+                    # <dt id="term-definition">definition</dt>
+                    # <dd>Text definition for the term</dd>
+                    # ...
+                    # </dl>
+                    parent = node.parent
+                    for n in parent.traverse():
+                        if n not in (node, node.next):
+                            n.remove()
+                    node = node.parent
+
+                elif 'citation' in node.parent.attributes.get('class'):
+                    # Sphinx HTML structure for sphinxcontrib-bibtex puts the ``id`` in the
+                    # ``dt`` element with the title of the cite. In this case, we
+                    # return the parent node which contains the definition list
+                    # and remove all ``dt/dd`` that are not the requested one
+
+                    # Structure:
+                    # <dl class="citation">
+                    # <dt id="cite-id"><span><a>Title of the cite</a></span></dt>
+                    # <dd>Content of the cite</dd>
+                    # ...
+                    # </dl>
+                    parent = node.parent
+                    for n in parent.traverse():
+                        if n not in (node, node.next):
+                            n.remove()
+                    node = node.parent
+
+                else:
+                    # Sphinx HTML structure for definition list puts the ``id``
+                    # the ``dt`` element, instead of the ``dl``. This makes
+                    # the backend to return just the title of the definition. If we
+                    # detect this case, we return the parent with the whole ``dl`` tag
+
+                    # Structure:
+                    # <dl class="confval">
+                    # <dt id="confval-config">
+                    # <code class="descname">config</code>
+                    # <a class="headerlink" href="#confval-config">¶</a></dt>
+                    # <dd><p>Text with a description</p></dd>
+                    # </dl>
+                    node = node.parent
+
+        return node.html
+
+    def get(self, request):
```

Comment:
Are we not doing any auth here?

Reply:
Not for now. Do we need auth at all? We are giving users to access only public internal projects (from inside `_get_page_content_from_storage` method) or public webpages on the internet.

I remember thinking about this and not finding a good and safe way to support private projects with session auth. Do we want other type of auth for this? If this is the case, we will need to consider auth _only_ for internal projects.

I thought it was safer, for now, to start only with public documents and grow from there if we require auth/private documents in the future.

Reply:
I think this can be implemented in corporate repository by overriding the class and the `permission_classes` field with a custom class that make these checks. Makes sense to you?

Reply:
Yea, seems reasonable.